src/main/java/org/folio/rs/service/RetrievalQueueService.java

```diff
@@ -93,4 +125,60 @@ private void saveRetrievalQueueWithCurrentDate(RetrievalQueueRecord record) {
     return (record, criteria, builder) -> builder.equal(record.get(ID), stringToUUIDSafe(id));
   }
 
+  private LocationMapping getLocationMapping(Item item) {
+    return locationMappingsService.getMappingByFolioLocationId(item.getEffectiveLocationId());
+  }
+
+  private Item getOriginalItemByBarcode(MovedEventRequest movedEventRequest) {
+    ResultList<Item> items = inventoryClient.getItem("barcode==" + movedEventRequest.getItemBarCode());
+    if (Objects.isNull(items)) {
+      throw new EntityNotFoundException("Item with barcode " + movedEventRequest.getItemBarCode() + NOT_FOUND);
+    }
+    return items.getResult().get(0);
+  }
+
+  private User getUserByRequesterId(MovedEventRequest movedEventRequest) {
+    ResultList<User> users = usersClient.query("id==" + movedEventRequest.getRequesterId());
+    if (Objects.isNull(users)) {
+      throw new EntityNotFoundException("User with id " + movedEventRequest.getRequesterId() + NOT_FOUND);
+    }
+    return users.getResult().get(0);
+  }
+
+  private Instance getInstanceByInstanceId(Item item) {
+    ResultList<Instance> instances = inventoryClient.getInstance("id==" + item.getInstanceId());
+    if (Objects.isNull(instances)) {
+      throw new EntityNotFoundException("Instance with id " + item.getInstanceId() + NOT_FOUND);
+    }
+    return instances.getResult().get(0);
+  }
+
+  private RetrievalQueueRecord buildRetrievalQueueRecord(MovedEventRequest movedEventRequest, Item item, User patron, LocationMapping mapping, Instance instance) {
+    return RetrievalQueueRecord.builder()
+        .holdId(movedEventRequest.getHoldId())
+        .patronBarcode(patron.getBarcode())
+        .patronName(patron.getUsername())
+        .callNumber(getCallNumber(item))
+        .itemBarcode(movedEventRequest.getItemBarCode())
+        .createdDateTime(LocalDateTime.now())
+        .pickupLocation(movedEventRequest.getPickupServicePointId())
+        .requestStatus(movedEventRequest.getRequestStatus())
+        .requestNote(movedEventRequest.getRequestNote())
+        .remoteStorageId(stringToUUIDSafe(mapping.getFolioLocationId()))
```

Comment:
Remote storage id should be used here.

Reply:
As i understand, we check location_mappings by effectiveLocationId of Item.If we have appropriate record in our db, that means - location is remote, so folioLocationId will be equal to remoteStorageId. Am i right?

Reply:
From what i see, folioLocationId is the same as effectiveLocationId and remote storage id is configuration id from location_mappings.